src/transformation/transformation.c

```diff
@@ -258,6 +310,148 @@ k4a_result_t transformation_2d_to_2d(const k4a_calibration_t *calibration,
+    if (K4A_FAILED(TRACE_CALL(transformation_create_depth_camera_pinhole(calibration, &pinhole))))
+    {
+        return K4A_RESULT_FAILED;
+    }
+
+    // Compute the 3d points in depth camera space that the current color camera pixel can be transformed to with the
+    // theoretical minimum and maximum depth values (mm)
+    float depth_range_mm[2] = { 50.f, 14000.f };
+    float start_point3d[3], stop_point3d[3];
+    int start_valid = 0;
+    if (K4A_FAILED(TRACE_CALL(transformation_2d_to_3d(calibration,
+                                                      source_point2d,
+                                                      depth_range_mm[0],
+                                                      K4A_CALIBRATION_TYPE_COLOR,
+                                                      K4A_CALIBRATION_TYPE_DEPTH,
+                                                      start_point3d,
+                                                      &start_valid))))
+    {
+        return K4A_RESULT_FAILED;
+    }
+
+    int stop_valid = 0;
+    if (K4A_FAILED(TRACE_CALL(transformation_2d_to_3d(calibration,
+                                                      source_point2d,
+                                                      depth_range_mm[1],
+                                                      K4A_CALIBRATION_TYPE_COLOR,
+                                                      K4A_CALIBRATION_TYPE_DEPTH,
+                                                      stop_point3d,
+                                                      &stop_valid))))
+    {
+        return K4A_RESULT_FAILED;
+    }
+
+    *valid = start_valid && stop_valid;
+    if (*valid == 0)
+    {
+        return K4A_RESULT_SUCCEEDED;
+    }
+
+    // Project above two 3d points into the undistorted depth image space with the pinhole model, both start and stop 2d
+    // points are expected to locate on the epipolar line
+    float start_point2d[2], stop_point2d[2];
+    start_point2d[0] = start_point3d[0] / start_point3d[2] * pinhole.fx + pinhole.px;
+    start_point2d[1] = start_point3d[1] / start_point3d[2] * pinhole.fy + pinhole.py;
+    stop_point2d[0] = stop_point3d[0] / stop_point3d[2] * pinhole.fx + pinhole.px;
+    stop_point2d[1] = stop_point3d[1] / stop_point3d[2] * pinhole.fy + pinhole.py;
+
+    // Search every pixel on the epipolar line so that its reprojected pixel coordinates in color image have minimum
+    // distance from the input color pixel coordinates
+    int depth_image_width_pixels = image_get_width_pixels(depth_image);
+    int depth_image_height_pixels = image_get_height_pixels(depth_image);
+    const uint16_t *depth_image_data = (const uint16_t *)(const void *)(image_get_buffer(depth_image));
+    float best_error = FLT_MAX;
+    float p[2];
+    p[0] = start_point2d[0];
+    p[1] = start_point2d[1];
+    float epipolar_line_slope = (stop_point2d[1] - start_point2d[1]) / (stop_point2d[0] - start_point2d[0]);
```

Comment:
>stop_point2d[0] - start_point2d[0] [](start = 72, length = 34)

In theory, that could lead to a division by 0.

Reply:
Maybe assert for vertical lines.

---
In reply to: [319257475](https://github.com/microsoft/Azure-Kinect-Sensor-SDK/pull/696#discussion_r319257475) [](ancestors = 319257475)